source/jormungandr/jormungandr/parking_space_availability/car/star.py

```diff
@@ -0,0 +1,118 @@
+#
+# LICENCE: This program is free software; you can redistribute it and/or modify
+# it under the terms of the GNU Affero General Public License as published by
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+# GNU Affero General Public License for more details.
+#
+# You should have received a copy of the GNU Affero General Public License
+# along with this program. If not, see <http://www.gnu.org/licenses/>.
+#
+# Stay tuned using
+# twitter @navitia
+# IRC #navitia on freenode
+# https://groups.google.com/d/forum/navitia
+# www.navitia.io
+from __future__ import absolute_import, print_function, unicode_literals, division
+
+import logging
+import pybreaker
+import requests as requests
+import jmespath
+
+from jormungandr import cache, app, utils, new_relic
+from jormungandr.parking_space_availability import AbstractParkingPlacesProvider
+from jormungandr.parking_space_availability.car.parking_places import ParkingPlaces
+from jormungandr.ptref import FeedPublisher
+
+DEFAULT_STAR_FEED_PUBLISHER = None
+
+
+class StarProvider(AbstractParkingPlacesProvider):
+
+    WS_URL_TEMPLATE = 'http://data.explore.star.fr/api/records/1.0/search/?dataset={}&refine.idparc={}'
+
+    def __init__(self, operators, dataset, timeout=1, feed_publisher=DEFAULT_STAR_FEED_PUBLISHER, **kwargs):
+        self.operators = [o.lower() for o in operators]
+        self.timeout = timeout
+        self.dataset = dataset
+
+        fail_max = kwargs.get('circuit_breaker_max_fail', app.config['CIRCUIT_BREAKER_MAX_STAR_FAIL'])
+        reset_timeout = kwargs.get('circuit_breaker_reset_timeout', app.config['CIRCUIT_BREAKER_STAR_TIMEOUT_S'])
+
+        self.breaker = pybreaker.CircuitBreaker(fail_max=fail_max, reset_timeout=reset_timeout)
+        self._feed_publisher = FeedPublisher(**feed_publisher) if feed_publisher else None
+
+        self.log = logging.LoggerAdapter(logging.getLogger(__name__), extra={'dataset': self.dataset})
+
+    def support_poi(self, poi):
+        properties = poi.get('properties', {})
+        return properties.get('operator', '').lower() in self.operators
+
+    def get_informations(self, poi):
+        ref = poi.get('properties', {}).get('ref')
+        if not ref:
+            return
+
+        data = self._call_webservice(ref)
+        if not data:
+            return
+
+        available = jmespath.search('records[0].fields.nombreplacesdisponibles', data) or 0
+        occupied = jmespath.search('records[0].fields.nombreplacesoccupees', data) or 0
+        # Person with reduced mobility
+        available_PRM = jmespath.search('records[0].fields.nombreplacesdisponiblespmr', data) or 0
+        occupied_PRM = jmespath.search('records[0].fields.nombreplacesoccupeespmr', data) or 0
```

Comment:
field not present and 0 must be different. If not present, that's maybe just that the API does not provide the information, not that there is no disabled slots.